src/ImportExport.ts

```diff
@@ -501,14 +501,15 @@ export const promocodes = async (input: string | null, amount?: number) => {
     if (
       Number.isNaN(toUse) ||
             !Number.isInteger(toUse) ||
-            toUse <= 0
+            toUse === 0 ||
+            (toUse < 0 && -toUse >= availableUses)
```

Comment:
if someone with 10 add uses puts in -10 it won't work, I don't understand the restriction. Not a major issue but I'd like to know the reasoning

Reply:
Using zero attempts was forbidden before this PR (toUse <= 0) and I didn't see a reason to change that. I wrote the validity check with that in mind.

Reply:
Wouldn't this simplification also be possible? Should check for both zero and negative values at once.
```suggestion
Number.isNaN(toUse) ||
    !Number.isInteger(toUse) ||
    -toUse >= availableUses
```

Reply:
I'm referring to someone entering -10 to use all of their hypothetical 10 add uses, which wouldn't work because -(-10) <= 10

Reply:
If I enter -10, that means I want to use all but 10 uses. If I only have 10 uses and enter -10, that means I want to save 10 out of 10 uses and use effectively none of them. If I want to use all of my 10 add uses, I'd put in 10 without a minus sign (which is also what the field comes pre-filled with then).

Reply:
that makes sense, wording confused me